### PR TITLE
Metrics debugging

### PR DESCRIPTION
**Pull request by jonas-schievink**

### CODE REVIEW

File: xtask/src/metrics.rs [hunk 1]
```diff
@@ -81,7 +82,11 @@ impl Metrics {
     }
     fn measure_analysis_stats_path(&mut self, name: &str, path: &str) -> Result<()> {
         eprintln!("\nMeasuring analysis-stats/{}", name);
-        let output = cmd!("./target/release/rust-analyzer analysis-stats --quiet {path}").read()?;
+        let output = Command::new("./target/release/rust-analyzer")
+            .args(&["analysis-stats", "--quiet", "--memory-usage", path])
+            .stdout(Stdio::piped())
+            .output()?;
```
Comment (matklad) [inline xtask/src/metrics.rs:88]: Wouldn't `output` pipe both stderr and stdout by default? I guess cmd! should have a flag to echo stderr...

Comment (jonas-schievink) [reply xtask/src/metrics.rs:88]: Ah, right. Yeah, this is to work around `cmd!` not having a matching API.

**Comment by matklad** [comment]
Semi related, field https://github.community/t/getting-access-to-perf-events-on-github-hosted-ubuntu-latest/149164 to ask around about perf insts.

**Comment by jonas-schievink** [comment]
not sure why it's passing on CI, let's see what happens on master

bors r+

File: xtask/src/metrics.rs (hunk 1 above)
Comment (matklad) [reply xtask/src/metrics.rs]: https://github.com/matklad/xshell/pull/11/commits/cf0657944cf0233457b0c150e1c85f5d04f10327

**Comment by jonas-schievink** [comment]
That fixed it?? Makes no sense

**Comment by jonas-schievink** [comment]
Ah, no, they're missing. But at least it no longer panics.

**Comment by jonas-schievink** [comment]
https://github.com/rust-analyzer/rust-analyzer/blob/24731e107f3c4790fc5fe73b63ad251a9d6e62ae/crates/rust-analyzer/src/lib.rs#L91

> thread 'main' panicked at 'Forgot to remove debug-print?', crates/rust-analyzer/src/lib.rs:91:9

**Comment by lnicola** [comment]
> Alternatively, is there perhaps a non-perf based way to measure how many CPU cycles my program spends?

Oof, Valgrind?

**Pull request merged**